components/centraldashboard/public/ajax_test_helper.js

```diff
@@ -26,3 +26,53 @@ export function mockRequest(component, response, respondWithError = false,
         component.addEventListener(responseEvent, () => resolve());
     });
 }
+
+/**
+ * Simply yeilds to the next tick, so that all Promised requests can do their
+ * thing.
+ */
+export async function yeildForRequests() {
+    const sleep = (t) => new Promise((res) => setTimeout(res, t));
```

Comment:
Why not just do await new Promise(..) and avoid defining the inner function?

Reply:
Changed it, since `sleep` is used in a few places.